### PR TITLE
feat:added option extension in the protobuf files

### PR DESCRIPTION
**Pull request by vishal-kanna**
## Description

closes: #3624 

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Comment by vishal-kanna** [comment]
@crodriguezvega  can you please look into the changes i made , if it is correct ,i'll proceed

**Comment by DimitrisJim** [comment]
the idea here is to use the option as defined by the sdk, not define one for ibc. I.e where applicable, bring in the option:

https://github.com/cosmos/cosmos-sdk/blob/025e02aeef88455032bc59904193ae449d5b805d/proto/cosmos/bank/v1beta1/tx.proto#L8

and then denote the signer:

https://github.com/cosmos/cosmos-sdk/blob/025e02aeef88455032bc59904193ae449d5b805d/proto/cosmos/bank/v1beta1/tx.proto#L23

hope this helps out!

**Comment by DimitrisJim** [comment]
hey @vishal-kanna, due to a recent PR which added some additional proto files (see the mentioned PR), we'd need to rebase and add the option to them too. 

To reduce the amount of rebasing you'd need to do, I'd suggest we wait until all PRs that add new messages are merged and then I'll ping you to add the option to the new messages too. How does that sound?

**Comment by vishal-kanna** [comment]
@DimitrisJim  it sounds great.Ping me once all the PRs are merged.

**Comment by crodriguezvega** [comment]
> I'd suggest we wait until all PRs that add new messages are merged and then I'll ping you to add the option to the new messages too.

I think I would prefer to merge this PR as is (I have rebased and added the option extension in the missing messages) and then add the extension in the PRs that already open for the migration of params. Otherwise, we might postpone merging this PR for too long.

**Comment by DimitrisJim** [comment]
Makes sense to me @crodriguezvega. Note that we probably do not want to close the original issue as of yet since we still have the new upgrade messages to annotate.

File: proto/ibc/applications/fee/v1/tx.proto
Comment (damiannolan) [reply proto/ibc/applications/fee/v1/tx.proto:10]: lol, I won't lose sleep over it

**Review by charleenfei — APPROVED** [review]

**Pull request merged**
